### PR TITLE
fix fluentbit ipv6 listen address

### PR DESCRIPTION
**Pull request by eumel8**
if logging-operator is deployed in IPv6 environment, fluentbit is not able to listen on http socket due the hard coded `HTTP_Listen` address. Make this dynamically set by `POD_IP` with an envVar definition

### CODE REVIEW

File: pkg/resources/fluentbit/config.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ var fluentBitConfigTemplate = `
     Coro_Stack_Size    {{ .CoroStackSize }}
     {{- if .Monitor.Enabled }}
     HTTP_Server  On
-    HTTP_Listen  0.0.0.0
```
Comment (eumel8) [inline pkg/resources/fluentbit/config.go:38]: another way: 

```
HTTP_Listen  [::]
```

works on IPv6 and doesn't need the Statefulset patch

**Comment by pepov** [comment]
I tried to rerun the e2e test multiple times, but it fails, haven't been able to dig deeper why. @eumel8 do you have an idea what can be the problem?

File: pkg/resources/fluentbit/config.go (hunk 1 above)
Comment (eumel8) [reply pkg/resources/fluentbit/config.go:38]: I mean the addional env var on $POD_IP in the Statefulset manifest. But this is not required anymore. Tested on Ipv6 and Ipv4 cluster the definition of `[::]` works.
The end2end test failed of the change in Statefulset manifest, but this should fixed now.

**Comment by csatib02** [comment]
I believe this PR is somewhat similiar to the one we merged not long ago: https://github.com/kube-logging/logging-operator/pull/1836

**Comment by eumel8** [comment]
@csatib02 Yes, after implementation we realized in an IPv6-only-cluster that fluentbit would listen only on IPv4 socket.

**Review by pepov — APPROVED** [review]

**Comment by withlin** [comment]
hi @eumel8  @pepov  shall we support set a enbale for ipv6 or ipv4? because somebody  do not support ipv6 yet  on machine for security.

**Comment by csatib02** [comment]
Hi @withlin,

Thank you for your feedback regarding IPv6 and IPv4 support for Fluentbit and Fluentd!

Adding the ability to configure this behavior at the operator level sounds like a reasonable enhancement. Contributions in this area are very welcome, as they would help address this use-case.

If you’re interested, feel free to open a pull request—we’d be happy to collaborate on it!

**Comment by eumel8** [comment]
Hi @withlin ,

yes, I had this in mind and [my previous approach](https://github.com/eumel8/logging-operator/compare/master...eumel8:logging-operator:fix/fluentbit-ipv6-2) was to use POD_IP instead of catch all listen address. Just tested on IPv4 cluster I've got this error

```
  Warning  FailedCreate      84s (x15 over 2m44s)  daemonset-controller  Error creating: pods "infra-fluentbit-" is forbidden: Get "https://[::1]:6443/api/v1/namespaces/infra/limitranges": dial tcp [::1]:6443: connect: cannot assign requested address
```

so, correct way would be to open issue and we can work on bug fix

**Pull request merged**